auto_compressor/src/lib.rs

```diff
@@ -50,3 +54,70 @@ impl FromStr for LevelInfo {
         Ok(LevelInfo(level_info))
     }
 }
+
+// PyO3 INTERFACE STARTS HERE
+#[pymodule]
+fn auto_compressor(_py: Python, m: &PyModule) -> PyResult<()> {
+    #[pyfn(m, compress_largest_rooms)]
+    fn compress_state_events_table(
+        py: Python,
+        db_url: String,
+        chunk_size: i64,
+        default_levels: String,
+        number_of_chunks: i64,
+    ) -> PyResult<()> {
+        // Stops the compressor from holding the GIL while running
+        py.allow_threads(|| {
+            compress_state_events_table_body(db_url, chunk_size, default_levels, number_of_chunks)
+        })
+    }
+
+    fn compress_state_events_table_body(
```

Comment:
I think it'd be cleaner for this function to live outside the `auto_compressor` function scope? Right now its a bit confusing as to whether its part of the module interface or not

Reply:
It has some python specific stuff (e.g. returning PyResult) so I didn't really want it accessible from outside of the python module. 

I could have the method start with an underscore (in a pythonesque way to say "you can't run me!")

I think it can't actually be run by anything at the moment (As it's not annotated with `#[pyfn(...)]`)

Reply:
Yeah, I mostly meant from a reading the code perspective. I would suggest putting it into its own module, but looks from the docs like things get weird if you try and do that. It's fine for now